src/simulators.jl

```diff
@@ -508,26 +508,160 @@ function simulate!(sys::ReplicaSystem{D, G, T},
                 "and simulators in TemperatureREMD ($(length(sim.simulators))) do not match."))
     end
 
+    if assign_velocities
+        for i in eachindex(sys.replicas)
+            random_velocities!(sys.replicas[i], sim.temperatures[i]; rng=rng)
+        end
+    end
+
+    simulate_remd!(sys, sim, n_steps, tremd_exchange!; rng=rng, n_threads=n_threads)
+        end
+
+function tremd_exchange!(sys::ReplicaSystem{D,G,T},
+                        sim::TemperatureREMD,
```

Comment:
```suggestion
                        sim::TemperatureREMD{T},
```

Should this be done (at other places in the file as well) in order to ensure that float type of system and simulator is same?

Reply:
At the moment we leave dealing with float types to the user and allow them to use any combination they want, so I think leave it out for now.